### PR TITLE
add ln-address-pay plugin

### PR DESCRIPTION
**Pull request by nosedam**
This is a simple plugin to make payments to lightning addresses as described in https://lightningaddress.com/

It just adds a new command to do 

`lightning-cli lnaddresspay address123@lightning.com 1000000`

I made it to top up my birefill account using the address they give you.

 I'm new to collaborating in open source projects so all help and feedback is appreciated

Repo url is: https://github.com/nosedam/ln-address-pay

Thanks!

### CODE REVIEW

**Comment by daywalker90** [comment]
you need to add this at the bottom (not the actual bottom we want these to be sorted alphabetically) of the README.md:
`[ln-address-pay]: https://github.com/nosedam/ln-address-pay`

**Comment by nosedam** [comment]
Oops really sorry, I forgot to add that change after forking
It should now be there, thanks!

**Comment by chrisguida** [comment]
This looks super cool!

**Comment by chrisguida** [comment]
Hmm, it seems the pip install of the dependencies is causing the CI to fail.

**Comment by nosedam** [comment]
Hi Chris thanks! Yeah I removes the secrets dependency. I'm going to check the permission issues now

**Comment by daywalker90** [comment]
```
ln_address_pay.py is not executable: Permission denied
```

it needs executable permissions:

```
chmod u+x ln_address_pay.py
```

**Comment by nosedam** [comment]
Haha Yeah I'm a filthy windows user. Thanks daywalker90! 

I uploaded it with the new permissions now, sorry for the troubles

**Comment by chrisguida** [comment]
Will merge this once someone tests it

**Comment by daywalker90** [comment]
I gave this plugin a quick test and the happy case seems to work fine. But if for example i use a fake address or an amount out of range i get an ugly formatted python error inside the json rpc error. I'd like to see some more user friendly errors that can be easier understood for beginner users.

E.g.:
```
lcli lnaddresspay odell@vlt.ge 1
{
   "code": -32600,
   "message": "Error while processing lnaddresspay: RpcError.__init__() missing 2 required positional arguments: 'payload' and 'error'",
   "traceback": "Traceback (most recent call last):\n  File \"/home/bitcoin/.local/lib/python3.11/site-packages/pyln/client/plugin.py\", line 641, in _dispatch_request\n    result = self._exec_func(method.func, request)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/bitcoin/.local/lib/python3.11/site-packages/pyln/client/plugin.py\", line 618, in _exec_func\n    ret = func(*ba.args, **ba.kwargs)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/bitcoin/ln-address-pay/ln_address_pay.py\", line 47, in pay_to_lightning_address\n    return RpcError(f\"Amount {msatoshi} is not within the allowed range\")\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nTypeError: RpcError.__init__() missing 2 required positional arguments: 'payload' and 'error'\n"
}
```

See how this error is even a little misleading at the end.

**Comment by nosedam** [comment]
Yeah I'm just returning an RPCError there so I totally agree on an improvement to make it more user friendly. I'm going to work on this! Thanks a lot!

**Comment by chrisguida** [comment]
Cool, I'll hold off on merging for now till the above problem is fixed. Thanks to both of you for your work so far!

**Comment by nosedam** [comment]
I added a response for errors that contains a status and a message to tell the user what failed in a more simple way

This is an example for an amount that is not within valid amounts reported by the lightning address server

image

Thanks for trying it and all feedback, errors look much better now

**Comment by chrisguida** [comment]
@nosedam Looks great to me! Would you be willing to squash this down to a single commit? I'll merge this once that's done!

**Comment by nosedam** [comment]
Oh sure! I squashed the commits into the first one. Thanks!

**Review by chrisguida — CHANGES_REQUESTED** [review]
One more nit

File: README.md
Comment (chrisguida) [inline README.md:219]: Can you move this down one line? It's not in alphabetical order if it's above listmempoolfunds.

Comment (nosedam) [reply README.md]: Done and squashed!

**Review by chrisguida — APPROVED** [review]
ACK https://github.com/lightningd/plugins/pull/572/commits/c735542f42f954855e013c803fa75af2a58c56bf

Thanks for the contribution! :heart:

**Comment by chrisguida** [comment]
Maybe add BOLT12 addresses next! ;)

https://github.com/bitcoin/bips/blob/master/bip-0353.mediawiki

**Comment by nosedam** [comment]
Yes I think thats a good followup and will look into it! Thanks for everything!

**Pull request merged**
